plugins/connection/network_cli.py

```diff
@@ -888,7 +910,22 @@ def send(
             response = self.receive(
                 command, prompt, answer, newline, prompt_retry_check, check_all
             )
-            return to_text(response, errors="surrogate_then_replace")
+            response = to_text(response, errors="surrogate_then_replace")
+
+            # check if the target device is in config mode
+            if use_cache and self.get_option("single_user_mode"):
```

Comment:
The value of `get_option("single_user_mode")`  can be set in class variable either in `connect` method and updated in the `update_play_context` to handle task level setting. This will prevent getting the option for each command run.